concepts/objects/about.md

```diff
@@ -0,0 +1,260 @@
+Even though the names must be strings, they do not need to be quoted if the names are "identifier-like" (composed of alphanumeric characters and underscore, and not started with a digit).
+
+```jq
+{name: "Jane", age: 42}
+```
+
+It is valid to use keys that are not "identifier-like".
+Just quote them.
+
+```jq
+{"first name": "Jane", "last name": "Lopez", age: 42}
+```
+
+If the name is the result of an expression, the expression **must** be in parentheses.
+
+```sh
+$ echo "Jane" | jq -Rc '{.: 42}'
+jq: error: syntax error, unexpected '.' (Unix shell quoting issues?) at <top-level>, line 1:
+{.: 42}
+jq: error: May need parentheses around object key expression at <top-level>, line 1:
+{.: 42}
+jq: 2 compile errors
+
+$ echo "Jane" | jq -Rc '{(.): 42}'
+{"Jane":42}
+```
+
+### Short-hand notations
+
+A couple of frequent cases for object construction have short-hand syntax.
+
+1. Re-using key-value pairs from an input object.
+
+   ```sh
+   $ jq -n '$ENV | {USER: .USER, EDITOR: .EDITOR}'
+   {
+     "USER": "glennj",
+     "EDITOR": "vim"
+   }
+   ```
+
+   `$ENV` is a object holding the name-value pairs of the environment.
+
+   The repetitive `key: .key` pairs can be written as just `key`
+
+   ```sh
+   $ jq -n '$ENV | {USER, EDITOR}'
+   {
+     "USER": "glennj",
+     "EDITOR": "vim"
+   }
+   ```
+		
+1. Similarly, there's a short-hand when using variables.
+
+   ```sh
+   $ jq -cn '"foo" as $a | "bar" as $b | {a: $a, b: $b}'
+   {"a":"foo","b":"bar"}
+
+   $ jq -cn '"foo" as $a | "bar" as $b | {$a, $b}'
+   {"a":"foo","b":"bar"}
+   ```
+
+## Indexing
+
+Values are retrieved from an object with _dot notation_.
+
+```jq
+{name: "Jane", age: 42} | .age    # => 42
+```
+
+If you cannot refer to the key as an identifier, use _bracket notation_.
+
+```jq
+"name" as $key | {name: "Jane", age: 42} | .$key    # => error
+"name" as $key | {name: "Jane", age: 42} | .[$key]  # => "Jane"
+```
+
+If the key is not in the object, the result is `null`
+
+```jq
+{name: "Jane", age: 42} | .sport  # => null
+```
+
+### Indexing nested values
+
+You will discover that this is an action you need to do again and again when parsing results from API queries.
+
+Given an object that contains array or object values, we can "chain" index expressions together.
+
+```jq
+{
+  "name": "Jane",
+  "age": 42,
+  "pets": ["cat", "fish"],
+  "address": {"street": "123 Main St", "city": "Springfield"}
+} as $example
+|
+  $example.pets[1],       # => "fish"
+  $example.address.city   # => "Springfield"
+```
+
+## Adding or changing key-value pairs
+
+Use the `=` assignment operator, with an index expression on the left-hand side.
+
+```jq
+{name: "Jane", age: 42} | .sport = "tennis"
+# => {
+#      "name": "Jane",
+#      "age": 42,
+#      "sport": "tennis"
+#    }
+```
+
+The `+` operator will _merge_ objects:
+
+```jq
+{Richard: 54} + {Jane: 42}
+# => {
+#      "Richard": 54,
+#      "Jane": 42
+#    }
+```
+
+## Removing a key
+
+Use the `del` function.
```

Comment:
```suggestion
Use the `del` function to remove a key.
```

A handful of sections start with sentences that are continuations of the header without being fully formed on their own.